File: LinkedTracker.Data/Models/Room.cs
using System.Collections.Generic;

namespace LinkedTracker.Data.Models;

public class Room((string game, string roomName) data)
{
    public string Game { get; set; } = data.game;
    public string RoomName { get; set; } = data.roomName;
    public string Password { get; set;}
    public string PointOfInterestType { get; set; }
    public List<PointOfInterest> PointsOfInterest { get; set; }
}

File: LinkedTracker.UI/wwwroot/js/room.js
// import poi from '../dist/js/PointOfInterest/POI.js'
//import Vue from 'vue'
import 'babel-polyfill';
import gameMap from '../../Components/GameMap.vue'

window.app = new Vue({
    el: '#app',
    data: {
        imgSrc: '/img/map2.png',
        ...window.vm
    },
    components: {
        gameMap
	},
    methods: {
        setPassword() {
            post('/Room/SetPassword', { 
                game: this.Room.Game,
                roomName: this.Room.RoomName,
                password: this.Room.Password
            })
        },
        async getPointsOfInterest() {
            var response = await fetch(`/POI/${this.Room.Game}/${this.Room.PointOfInterestType}`)
            this.PointsOfInterest = await response.json()
        }
    }
})